.github/workflows/ci.yml

```diff
@@ -0,0 +1,75 @@
+name: CI
+
+on:
+  pull_request:
+    branches:
+      - main
+
+env:
+  HF_SCRIPTS_VERSION: main
+  HF_ALLOW_CODE_EVAL: 1
+
+jobs:
+
+  check_code_quality:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.6"
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install .[quality]
+      - name: Check quality
+        run: |
+          black --check --line-length 119 --target-version py36 tests src benchmarks datasets metrics
+          isort --check-only tests src benchmarks datasets metrics
+          flake8 tests src benchmarks datasets metrics
+
+  test:
+    needs: check_code_quality
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+        pyarrow_version: [latest, 6.0.1]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Install OS dependencies
+        if: ${{ matrix.os == 'ubuntu-latest' }}
+        run: |
+          sudo apt-get -y update
+          sudo apt-get -y install libsndfile1-dev
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
```

Comment:
Note `fetch-depth: 0` means to fetch all history commits for all branches and tags (instead of default, just the merge commit of current branch to main branch).

The reason for this is to be able to collect the test `tests/test_dataset_cards.py::test_changed_dataset_card`, because these are parametrized with the function `get_changed_datasets` and its implementation requires both current branch and origin/main branch: `origin/main...HEAD`
```python
diff_output = check_output(["git", "diff", "--name-only", "origin/main...HEAD"], cwd=repo_path)
```

Otherwise, an error is raised:
```
_________________ ERROR collecting tests/test_dataset_cards.py _________________
tests/test_dataset_cards.py:53: in <module>
    @pytest.mark.parametrize("dataset_name", get_changed_datasets(repo_path))
tests/test_dataset_cards.py:35: in get_changed_datasets
    diff_output = check_output(["git", "diff", "--name-only", "origin/main...HEAD"], cwd=repo_path)
/opt/hostedtoolcache/Python/3.6.15/x64/lib/python3.6/subprocess.py:356: in check_output
    **kwargs).stdout
/opt/hostedtoolcache/Python/3.6.15/x64/lib/python3.6/subprocess.py:438: in run
    output=stdout, stderr=stderr)
E   subprocess.CalledProcessError: Command '['git', 'diff', '--name-only', 'origin/main...HEAD']' returned non-zero exit status 128.
```
beacuase of:
```
fatal: ambiguous argument 'origin/main...HEAD': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
```